### PR TITLE
[ADD] modules: look recursively for modules in addons path

### PR DESCRIPTION
**Pull request by beledouxdenis**

### CODE REVIEW

**Comment by Yenthe666** [comment]
@beledouxdenis is my assumption correct that this means we can simply add an addons path in the configuration file saying `/opt/custom` and that all subfolders containing modules will be auto-loaded? Such as `/opt/custom/repo1` and `/opt/custom/repo2`? 
How would that turn out if you migrate a customer from V11 to V12 and forget to change the conf file? I'm curious.

File: odoo/modules/module.py [hunk 1]
```diff
@@ -153,6 +154,21 @@ def initialize_sys_path():
...
 
+
+def is_addons_path(path, recursive=True):
+    for path, dirnames, _f in os.walk(path):
+        for f in dirnames:
+            modpath = os.path.join(path, f)
+            if os.path.isdir(modpath):
+                def hasfile(filename):
+                    return os.path.isfile(os.path.join(modpath, filename))
+                if hasfile('__init__.py') and any(hasfile(mname) for mname in MANIFEST_NAMES):
```
Comment (xmo-odoo) [inline odoo/modules/module.py]: Not sure about that one for support of data-only addon, but I don't really know.

Comment (beledouxdenis) [reply odoo/modules/module.py]: This is a simple move of the method from one file to another.
See
https://github.com/odoo/odoo/pull/27217/files#diff-3813bc998e290bd756b27aef89662e82L495

**Comment by beledouxdenis** [comment]
It was decided to not add this feature after-all.

**Pull request closed without merging**